### DISCUSSION TITLE
How to retrieve 'LayeredExtremum' of material properties?

### DISCUSSION THREAD

**Discussion by slee-USNC** [idaholab/moose#24672] [Q&A General] [answered]
Can we retrieve layer-wise extremum of material properties? I am trying to extract maximum temperature of specific material within different intervals on the z-axis. 

e.g.) Given a mesh like below:
```
[Mesh]
  [hexagon]
    type = PolygonConcentricCircleMeshGenerator
    num_sides = 6
    num_sectors_per_side = '4 4 4 4 4 4'
    ring_intervals = '1 1 1 1'
    ring_radii = '1 1 1 1'
    ring_block_ids = '1 2 3 4'
    polygon_size = 6
    background_block_ids = 5
    quad_center_elements = True
  []
  [extrusion]
    type = AdvancedExtruderGenerator
    input = hexagon
    direction = '0 0 1'
    heights = '1 1 1 1 1'
    num_layers = '1 1 1 1 1'
  []
[]
```
I'd like to get the max temperature for every 1 m axial layer of block 3.

I know [`LayeredAverage`](https://mooseframework.inl.gov/source/userobjects/LayeredAverage.html) does something similar, but I want to retrieve maximum and minimum instead of the average.

**Comment by GiudGiud** [comment, marked as answer]
Hello

We need to create a LayeredExtrema class to do this. It will be able to do min/max and will look much like the LayeredAverage code
Do you feel comfortable creating it?
This is something that could be contributed to the framework.

Guillaume

**Reply by slee-USNC** [reply]
> Do you feel comfortable creating it?

I have to say no unfortunately. I don't have any experience in MOOSE development yet.

**Reply by GiudGiud** [reply]
What kind of timeline do you have on this?
Are you working in collaboration with a program at INL or ANL on this?
Feel free to email me at guillaume.giudicelli@inl.gov if this is confidential

**Reply by slee-USNC** [reply]
> What kind of timeline do you have on this?

We don't need this in urgent. Definitely something to explore in the future though.

>  Are you working in collaboration with a program at INL or ANL on this?

No

**Reply by GiudGiud** [reply]
I've created this issue to keep track of this feature request
https://github.com/idaholab/moose/issues/24682 

I also made this branch:
https://github.com/GiudGiud/moose/tree/PR_extrema
which has a LayeredExtremumMaterialProperty user object
I did not test it, and I have to move on to something else rn, please let me know if it works as expected.
I ll try to test it this week

**Reply by slee-USNC** [reply]
@GiudGiud 
Thank you! We will try the new branch out on our end as well

**Reply by GiudGiud** [reply]
Trying it right now too. looks like I duplicated a parameter, I ll do some clean up

**Reply by GiudGiud** [reply]
yeah it does not work yet. I ll see if I get it working soon.
Sorry for the false hope

**Reply by GiudGiud** [reply]
Should work now. I got a reasonable result for my test input.

Please let me know what you see on your end

I pushed a new version to the branch with my test too

**Reply by slee-USNC** [reply]
It seems working well and as expected on our end too. Thanks @GiudGiud !

**Reply by GiudGiud** [reply]
Good to hear.
I just improved my test a little and saw an issue with every other value when outputting through a SpatialUOVPP. Just letting you know

**Reply by GiudGiud** [reply]
oh it s because I was sampling 10 layers of maximum with only 6 elements. nvmd

**Comment by GiudGiud** [comment]
LayeredExtremumMaterialProperty is merged into MOOSE, in the next/ branch. It will propagate in devel / master then in the applications over the next few days
